### ISSUE TITLE
ParallelSSHClient (proxy): ssh key authentication failed without pkey and proxy_key parameters

### ISSUE THREAD

**Issue by fsaintma** [ParallelSSH/parallel-ssh#245]
### Bug reports

__Steps to reproduce:__

1. Example code that produces error.

``` 
from pssh.clients import ParallelSSHClient
import logging
from pssh import logger

logger.setLevel(logging.DEBUG)
logging.basicConfig()

client = ParallelSSHClient(['host1'],
                           user='root',
                           proxy_host='host2',
                           proxy_user='user2'')
output = client.run_command('uptime', stop_on_errors=False)
client.join(output)
for host in output:
    print(host)
    if host.stdout is not None:
        for line in host.stdout:
            print(line)
```
                         

2. Stack trace or error messages.

user2, host2, host1 are fake values and I replace them in the stacktrace for the issue

```
DEBUG:pssh.clients.base.parallel:_run_command with read timeout None
DEBUG:pssh.clients.native.parallel:Make client request for host host1, (host_i, host) in clients: False
DEBUG:pssh.clients.base.single:Connecting to host2:22
DEBUG:pssh.clients.native.single:Agent auth failed with Unable to connect to agentcontinuing with other authentication methods
DEBUG:pssh.clients.base.single:Trying to authenticate with identity file /home/users/user2/.ssh/id_rsa
DEBUG:pssh.clients.base.single:Authentication with identity file  /home/users/user2/.ssh/id_rsa failed, continuing with other identities
DEBUG:pssh.clients.base.single:Trying to authenticate with identity file /home/users/user2/.ssh/id_dsa
DEBUG:pssh.clients.base.single:Authentication with identity file/home/users/user2/.ssh/id_dsa failed, continuing with other identities
DEBUG:pssh.clients.base.single:Trying to authenticate with identity file /home/users/user2/.ssh/id_ecdsa
DEBUG:pssh.clients.base.single:Authentication with identity file /home/users/user2/.ssh/id_ecdsa failed, continuing with other identities
DEBUG:pssh.clients.base.single:Trying to authenticate with identity file /home/users/user2/.ssh/id_ed25519
DEBUG:pssh.clients.base.single:Authentication with identity file /home/users/user2/.ssh/id_ed25519 failed, continuing with other identities
DEBUG:pssh.clients.native.single:Agent auth failed with Unable to connect to agentcontinuing with other authentication methods

....
ERROR:pssh.clients.base.parallel:Failed to run on host None - ('Proxy authentication failed. Exception from tunnel client: %s', AuthenticationError('Authentication error while connecting to %s:%s - %s', 'host2', 22, AuthenticationError('No authentication methods succeeded')))
	host=None
	exit_code=None
	channel=None
	stdout=None
	stderr=None
	stdin=None
	exception=('Proxy authentication failed. Exception from tunnel client: %s', AuthenticationError('Authentication error while connecting to %s:%s - %s', 'host2', 22, AuthenticationError('No authentication methods succeeded')))
	encoding=utf-8
	read_timeout=None
```

__Expected behaviour:__

I'm expecting to see the output of uptime command on the host `host1`.

__Actual behaviour:__

The key authentication failed. If I add in ParallelSSHClient object the parameters  `pkey='~/.ssh/id_rsa'` and `proxy_pkey='~/.ssh/id_rsa'` (i.e. specify manually the path of the private key) it works normally. You can see the stack trace below. I suppose that by not specifying the keys I should get the same result.

```
DEBUG:pssh.clients.base.parallel:_run_command with read timeout None
DEBUG:pssh.clients.native.parallel:Make client request for host host1, (host_i, host) in clients: False
DEBUG:pssh.clients.base.single:Connecting to host2:22
DEBUG:pssh.clients.native.single:Proceeding with private key file authentication
DEBUG:pssh.clients.base.single:Authentication completed successfully - setting session to non-blocking mode
DEBUG:pssh.clients.native.tunnel:Hub in server runner is main hub: False
DEBUG:pssh.clients.base.single:Connecting to 127.0.0.1:43743
DEBUG:pssh.clients.native.tunnel:Client connected, forwarding host1:22 on remote host to 127.0.0.1:50450

DEBUG:pssh.clients.base.single:Reading from stderr buffer, timeout=None
	host=host1
	exit_code=0
	channel=<ssh2.channel.Channel object at 0x7f5d2d73f250>
	stdout=<generator object BaseSSHClient.read_output_buffer at 0x7f5d3166e5f0>
	stderr=<generator object BaseSSHClient.read_output_buffer at 0x7f5d30999200>
	stdin=<ssh2.channel.Channel object at 0x7f5d2d73f250>
	exception=None
	encoding=utf-8
	read_timeout=NoneDEBUG:pssh.clients.native.tunnel:Wrote all data to channel

DEBUG:pssh.clients.base.single:Reading from stdout buffer, timeout=None
DEBUG:pssh.clients.base.single:Reading from stdout buffer, timeout=None
INFO:pssh.host_logger:[host1]	 12:02:39 up  3:25,  0 users,  load average: 0.00, 0.01, 0.05
 12:02:39 up  3:25,  0 users,  load average: 0.00, 0.01, 0.05
```

__Additional info:__ [Include version of ``ssh2-python`` and/or ``paramiko`` and any other relevant information.]

parallel-ssh: **2.3.1**
paramiko: **2.7.2**
ssh2-python: **0.25.0**

**Comment by pkittenis** [comment]
Hello,

Thank you for the interest.

The library uses an available SSH agent for authentication if neither password nor private key are provided, as well as default identities under user home directory.

An agent was not available on your system and no other authentication mechanisms were provided. Default identity files all failed.

> DEBUG:pssh.clients.native.single:Agent auth failed with Unable to connect to agentcontinuing with other authentication methods
> DEBUG:pssh.clients.base.single:Trying to authenticate with identity file /home/users/user2/.ssh/id_rsa
> DEBUG:pssh.clients.base.single:Authentication with identity file  /home/users/user2/.ssh/id_rsa failed, continuing with other identities
> DEBUG:pssh.clients.base.single:Trying to authenticate with identity file /home/users/user2/.ssh/id_dsa
> DEBUG:pssh.clients.base.single:Authentication with identity file/home/users/user2/.ssh/id_dsa failed, continuing with other identities
> DEBUG:pssh.clients.base.single:Trying to authenticate with identity file /home/users/user2/.ssh/id_ecdsa
> DEBUG:pssh.clients.base.single:Authentication with identity file /home/users/user2/.ssh/id_ecdsa failed, continuing with other identities
> DEBUG:pssh.clients.base.single:Trying to authenticate with identity file /home/users/user2/.ssh/id_ed25519
> DEBUG:pssh.clients.base.single:Authentication with identity file /home/users/user2/.ssh/id_ed25519 failed, continuing with other identities

At least one authentication mechanism must be available or authentication will fail.

The above is expected behaviour.

See [documentation for more examples](https://parallel-ssh.readthedocs.io/en/latest/quickstart.html#run-a-command-on-hosts-in-parallel).

**Comment by pkittenis** [comment]
This was a duplicate of #222 - resolved in `2.5.x`.
